spec/Candid.md

```diff
@@ -464,14 +464,26 @@ A *function reference* is described by its function type. For example, they allo
 <reftype> ::= func <functype> | ...
 ```
 
-##### Example
+A function reference may be represented by a simple closure, encapsulating a prefix of arguments that have previously been bound. These hidden arguments do not appear in the function type, but will be forwarded implicitly when the function itself is invoked.
+
+
+##### Examples
 
 ```
 type engine = service {
   search : (query : text, callback : func (vec result) -> ());
 }
 ```
 
+```
+type wallet = service {
+  topup : (amount : nat) -> ();
+  forward : (call : () -> ()) -> ();
```

Comment:
```suggestion
  forward : (call : func () -> ()) -> ();
```
syntactically, how do we know if the function has bound arguments? Does the closure supports return type?

Reply:
You mean, know it by the type? We don't, it's always allowed. Do you think that would cause issues for bindings?

Reply:
I see. The existing example has a `callback` function, which can have bound arguments as well. You are adding another example to illustrate the use of bound arguments, which makes me think there may be syntactic difference for the bound args.

So any function reference can have bound arguments. The return value will be decoded according to the function signature, right? That means we need an abstraction for argument tuples. And if we use the `dynamic` type as return type, `dynamic` represents argument tuples instead of a single argument.

Reply:
> That means we need an abstraction for argument tuples.

Yep.  That's my thinking too, and more agreement in the comments for #292